frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -490,13 +490,16 @@ const GitSection: React.FC<GitSectionProps> = ({
         helpText={helpText}
         helpTextInvalid={helpText}
         validated={validated}
-        onChange={(e: React.SyntheticEvent) => {
+        onBlur={(e: { target: { value: string } }) => {
+          const trimmedURL = e.target.value.trim();
+          setFieldValue('git.url', trimmedURL);
+          if (e.target.value !== trimmedURL) {
+            debouncedHandleGitUrlChange(trimmedURL, values.git.ref, values.git.dir);
+          }
```

Comment:
Thanks, generally looks good. Just have one more suggestion:
```suggestion
          if (e.target.value !== trimmedURL) {
            setFieldValue('git.url', trimmedURL);
            debouncedHandleGitUrlChange(trimmedURL, values.git.ref, values.git.dir);
          }
```
Would be nice if we set value only if it has changed.